testing/scripts/playwright_test.py

```
@@ -1946,6 +1946,299 @@ async def test_api_v220_rate_limit_contract(page: Page) -> None:
+async def test_eslint_clean(page: Page) -> None:  # noqa: ARG001
+    """Run ESLint and assert exit code 0."""
+    import shutil
+    import subprocess
+    from pathlib import Path
+    section("ESLint — app.js clean")
+    npm = shutil.which("npm")
+    if npm is None:
+        assert_true("npm is on PATH", False, "npm not found — cannot run ESLint")
+        return
+    repo_root = Path(__file__).resolve().parents[2]
+    result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)